app/src/sharedTest/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentTest.kt

```diff
@@ -0,0 +1,427 @@
+          getFeatureFlagDisplayName(ephemeralFeatureFlag.id)
+        )
+        verifyFeatureFlagSyncStatus(
+          index,
+          getSyncStatusText(ephemeralFeatureFlag.syncStatus)
+        )
+        verifyFeatureFlagSwitchState(
+          index,
+          ephemeralFeatureFlag.currentValue
+        )
+      }
+    }
+  }
+
+  @Test
+  fun testFeatureFlagFragment_downloadSupportFlag_hasCorrectDetails() {
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      val downloadSupportFlag = getEphemeralFeatureFlags()[0]
+
+      scrollToPosition(0)
+      verifyFeatureFlagDisplayName(
+        0,
+        getFeatureFlagDisplayName(downloadSupportFlag.id)
+      )
+      verifyFeatureFlagSyncStatus(
+        0,
+        getSyncStatusText(downloadSupportFlag.syncStatus)
+      )
+      verifyFeatureFlagSwitchState(
+        0,
+        downloadSupportFlag.currentValue
+      )
+    }
+  }
+
+  @Test
+  fun testFeatureFlagFragment_downloadSupportFlag_switchToggled_updatesValue() {
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      val downloadSupportFlag = getEphemeralFeatureFlags()[0]
+
+      scrollToPosition(0)
+      onView(
+        atPositionOnView(
+          R.id.feature_flags_recycler_view,
+          0,
+          R.id.feature_flag_switch
+        )
+      ).perform(click())
+
+      verifyFeatureFlagSwitchState(
+        0,
+        !downloadSupportFlag.currentValue
+      )
+    }
+  }
+
+  @Test
+  fun testFeatureFlagFragment_toggleDownloadSupportFlag_configChanges_valuePersists() {
+    launch(FeatureFlagsTestActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+      val downloadSupportFlag = getEphemeralFeatureFlags()[0]
+
+      scrollToPosition(0)
+      onView(
+        atPositionOnView(
+          R.id.feature_flags_recycler_view,
+          0,
+          R.id.feature_flag_switch
+        )
+      ).perform(click())
+
+      verifyFeatureFlagSwitchState(
+        0,
+        !downloadSupportFlag.currentValue
+      )
+
+      onView(isRoot()).perform(OrientationChangeAction.orientationLandscape())
+
+      verifyFeatureFlagSwitchState(
+        0,
+        !downloadSupportFlag.currentValue
+      )
+    }
+  }
+
+  private fun verifyFeatureFlagDisplayName(
+    position: Int,
+    expectedDisplayName: String
+  ) {
+    onView(
+      atPositionOnView(
+        R.id.feature_flags_recycler_view,
+        position,
+        R.id.feature_flag_label_text_view
+      )
+    ).check(
+      matches(
+        withText(expectedDisplayName)
+      )
+    )
+  }
+
+  private fun verifyFeatureFlagSyncStatus(
+    position: Int,
+    expectedSyncStatus: String
+  ) {
+    onView(
+      atPositionOnView(
+        R.id.feature_flags_recycler_view,
+        position,
+        R.id.sync_status_value_text_view
+      )
+    ).check(
+      matches(
+        withText(expectedSyncStatus)
+      )
+    )
```

Comment:
Here and above: This can be fit in a single line?
```suggestion
    ).check(matches(withText(expectedSyncStatus)))
```

Reply:
updated